app/data/Owners.scala

```diff
@@ -0,0 +1,37 @@
+package data
+
+import model._
+
+trait Owners {
+  def stacks: Set[(String, SSA)]
+
+  def all: Set[Owner] = stacks
+    .groupBy(_._1)
+    .map { case (ownerId, stacks) =>
+      Owner(ownerId, stacks.map(_._2))
+    }.toSet
+
+  def find(id: String): Option[Owner] = all.find(_.id == id)
+
+  def find(ssa: SSA): Option[Owner] = all.find(_.hasSSA(ssa))
+
+  def forStack(stackName: Option[String], stageName: Option[String], appName: Option[String]): Option[Owner] = all
+    .find(_.hasSSA(SSA(stackName, stageName, appName)))
+    .orElse(all.find(_.hasSSA(SSA(stackName, stageName))))
+    .orElse(all.find(_.hasSSA(SSA(stackName))))
+    .orElse(all.find(_.hasSSA(SSA())))
+}
```

Comment:
I think there are some potentially weird behaviours allowed here. For example providing a `stageName` or an `appName` only (without a `stackName`) would return the default owner. I would argue that this is an error and shouldn't return a result at all.

I think the legal queries are:
 - stack only
 - stack and stage
 - stack and app
 - stack, stage and app

Any other combination seems to make little sense and should perhaps return no owner or an error?